### PR TITLE
Add New*CRD() functions for Cluster API types & register Cluster CRD

### PR DESCRIPTION
**Pull request by tuommaki**
To get OperatorKit create CRD for defined type, there must be object
representation of that CRD passed to it. This change makes Cluster controller
to register Cluster CRD and provides preliminary version for MachineDeployment
as well.

### CODE REVIEW

File: service/controller/clusterapi/crds.go [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+//       scope: Namespaced
+//       version: v1alpha1
+//       names:
+//         kind: Cluster
+//         plural: clusters
+//         singular: cluster
+//       subresources:
+//         status: {}
+//
+func NewClusterCRD() *apiextensionsv1beta1.CustomResourceDefinition {
```
Comment (xh3b4sd) [inline service/controller/clusterapi/crds.go]: Why don't we have this in `apiextensions`? I think it is wrong here. Imagine we introduce the Cluster API types in Azure and KVM, what then?

Comment (tuommaki) [reply service/controller/clusterapi/crds.go]: I was thinking about that, but I felt it was wrong to have these in `apiextensions` when the types aren't defined there. Which package would you put them there? I'm fine moving these but I thought initially that it's not semantically right.

Comment (kopiczko) [reply service/controller/clusterapi/crds.go]: I agree this should land in `apiextensions`. My suggestion is `cluster/v1alpha1`. Even though we don't have types there we still have provider extensions.

**Comment by tuommaki** [comment]
Moving to `apiextensions`

**Pull request closed without merging**